lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -1037,30 +1037,12 @@ ParseResult FIRParser::parseOptionalRUW(RUWAttr &result) {
 //===----------------------------------------------------------------------===//
 
 // Entries in a symbol table are either an mlir::Value for the operation that
-// defines the value or an unbundled ID tracking the index in the
-// UnbundledValues list.
-using UnbundledID = llvm::PointerEmbeddedInt<unsigned, 31>;
-using SymbolValueEntry = llvm::PointerUnion<Value, UnbundledID>;
-
+// defines the value or an operation.
+using SymbolValueEntry = llvm::PointerUnion<Value, Operation *>;
 using ModuleSymbolTable =
     llvm::StringMap<std::pair<SMLoc, SymbolValueEntry>, llvm::BumpPtrAllocator>;
 using ModuleSymbolTableEntry = ModuleSymbolTable::MapEntryTy;
 
-using UnbundledValueEntry = SmallVector<std::pair<Attribute, Value>>;
```

Comment:
Could this be done by keeping this around and having the ID point to the Operation + the name/value pairs for the "unbundled" bits?  Not really suggesting that necessarily, but that's what's needed here-- the ability to resolve to the operation itself using its name, right?

This does make the resolution dependent on the generated operation's attributes (vs parser state, also a bit less general although don't think that matters) but as discussed we do that elsewhere already so :man_shrugging: .